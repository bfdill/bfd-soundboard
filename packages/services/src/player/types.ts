import { ChildProcess } from 'child_process'

export type DefaultPlayer =
  | 'mplayer'
  | 'afplay'
  | 'mpg123'
  | 'mpg321'
  | 'play'
  | 'omxplayer'
  | 'aplay'
  | 'cmdmp3'
export type What = string | string[]
export type PlayerOptions = {
  player: any
  players?: DefaultPlayer | DefaultPlayer[] | string[]
}
export type PlayerArgs = { [playerName: string]: string[] }
export type PlayOptions = PlayerArgs & { stdio: 'ignore' }
export type Next = Function
export interface IPlayer {
  play: (what: What, options?: PlayOptions | Next, next?: Next) => ChildProcess
}
export type createPlayer = (opts?: PlayerOptions) => IPlayer
