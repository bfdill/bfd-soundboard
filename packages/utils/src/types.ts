export type DeepPartial<T> = {
  [P in keyof T]?: T[P] extends (infer U)[]
    ? DeepPartial<U>[]
    : T[P] extends ReadonlyArray<infer U>
    ? ReadonlyArray<DeepPartial<U>>
    : T[P] extends (string | number | boolean)
    ? T[P]
    : DeepPartial<T[P]>
}
