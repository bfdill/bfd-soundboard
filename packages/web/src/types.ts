import { Action } from 'redux'
import { ThunkAction } from 'redux-thunk'

export type FullState = {
  sounds: any
}

export type RequestGetSoundsAction = Action<'REQUEST/getSounds'>
export type SuccessGetSoundsAction = Action<'SUCCESS/getSounds'> & {
  payload: any
}
export type FailGetSoundsAction = Action<'FAILED/getSounds'> & { err: Error }
export type AnyAction =
  | RequestGetSoundsAction
  | SuccessGetSoundsAction
  | FailGetSoundsAction
export type GetSounds = () => ThunkAction<
  void,
  FullState,
  undefined,
  RequestGetSoundsAction | SuccessGetSoundsAction | FailGetSoundsAction
>
