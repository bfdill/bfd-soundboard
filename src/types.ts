import * as Koa from 'koa'
import * as KoaRouter from 'koa-router'
import { Server } from 'http'

export interface IApplication {
  listen: (port: number) => Server
}

export type ApplicationState = {
  correlationId: string
}
export interface IApplicationContext
  extends Koa.ParameterizedContext<ApplicationState> {}
export type ApplicationMiddleware = Koa.Middleware<
  ApplicationState,
  IApplicationContext
>
export type ApplicationKoa = Koa<ApplicationState, IApplicationContext>
export type ApplicationRouter = KoaRouter<ApplicationState, IApplicationContext>
