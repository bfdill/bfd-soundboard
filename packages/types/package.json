{
  "name": "@talkboy-9000/types",
  "version": "0.0.1"
}
