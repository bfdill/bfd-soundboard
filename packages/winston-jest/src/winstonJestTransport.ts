import * as winstonTransport from 'winston-transport'
import { IWinstonJestTransport } from './winstonJestTransport.types'

export class WinstonJestTransport extends winstonTransport
  implements IWinstonJestTransport {
  public readonly mock: jest.Mock = jest.fn()

  constructor(opts?: winstonTransport.TransportStreamOptions) {
    super(opts)
  }

  log = (info: any, next: () => void): any => {
    this.mock(info)
    next()
  }

  logv = (info: any, next: () => void): any => {
    this.mock(info)
    next()
  }

  callsMatchSnapshot = () =>
    this.mock.mock.calls.forEach(call => expect(call).toMatchSnapshot())
}
