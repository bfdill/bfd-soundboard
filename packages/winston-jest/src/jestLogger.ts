import * as winston from 'winston'
import { IJestLogger } from './jestLogger.types'
import { WinstonJestTransport } from './winstonJestTransport'

export const snapshotExistingLogger = (logger: winston.Logger) => {
  const transport = new WinstonJestTransport()
  logger.clear()
  logger.add(transport)
  logger.info('✅test works✅')
  transport.callsMatchSnapshot()
}

export const getJestLogger = (): IJestLogger => {
  const transport: WinstonJestTransport = new WinstonJestTransport()
  return {
    transport,
    callsMatchSnapshot: () => transport.callsMatchSnapshot(),
    logger: winston.createLogger({
      level: 'silly',
      levels: winston.config.npm.levels,
      transports: [transport]
    })
  }
}
