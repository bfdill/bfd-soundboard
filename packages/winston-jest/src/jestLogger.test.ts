import { IJestLogger } from './jestLogger.types'
import { getJestLogger } from './jestLogger'
import { WinstonJestTransport } from './winstonJestTransport'

describe('getJestLogger', () => {
  let sut: IJestLogger

  beforeEach(() => {
    sut = getJestLogger()
  })

  test('transport', () => {
    expect(sut.transport).toBeDefined()
    expect(sut.transport instanceof WinstonJestTransport).toBeTruthy()
  })

  test('logger', () => {
    expect(sut.logger).toBeDefined()
    // meh, tmi
    expect(sut.logger.transports[0]).toBe(sut.transport)
  })

  test('callsMatchSnapshot', () => {
    expect(sut.callsMatchSnapshot).toBeDefined()

    sut.logger.log('error', 'call 1')
    sut.logger.log('warn', 'call 2')
    sut.logger.log('info', 'call 3')
    sut.logger.log('verbose', 'call 4')
    sut.logger.log('debug', 'call 5')
    sut.logger.log('silly', 'call 6')

    sut.callsMatchSnapshot()
  })
})
