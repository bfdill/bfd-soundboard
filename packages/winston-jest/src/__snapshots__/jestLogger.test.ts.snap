// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getJestLogger callsMatchSnapshot 1`] = `
Array [
  Object {
    "level": "error",
    "message": "call 1",
    Symbol(level): "error",
    Symbol(message): "{\\"level\\":\\"error\\",\\"message\\":\\"call 1\\"}",
  },
]
`;

exports[`getJestLogger callsMatchSnapshot 2`] = `
Array [
  Object {
    "level": "warn",
    "message": "call 2",
    Symbol(level): "warn",
    Symbol(message): "{\\"level\\":\\"warn\\",\\"message\\":\\"call 2\\"}",
  },
]
`;

exports[`getJestLogger callsMatchSnapshot 3`] = `
Array [
  Object {
    "level": "info",
    "message": "call 3",
    Symbol(level): "info",
    Symbol(message): "{\\"level\\":\\"info\\",\\"message\\":\\"call 3\\"}",
  },
]
`;

exports[`getJestLogger callsMatchSnapshot 4`] = `
Array [
  Object {
    "level": "verbose",
    "message": "call 4",
    Symbol(level): "verbose",
    Symbol(message): "{\\"level\\":\\"verbose\\",\\"message\\":\\"call 4\\"}",
  },
]
`;

exports[`getJestLogger callsMatchSnapshot 5`] = `
Array [
  Object {
    "level": "debug",
    "message": "call 5",
    Symbol(level): "debug",
    Symbol(message): "{\\"level\\":\\"debug\\",\\"message\\":\\"call 5\\"}",
  },
]
`;

exports[`getJestLogger callsMatchSnapshot 6`] = `
Array [
  Object {
    "level": "silly",
    "message": "call 6",
    Symbol(level): "silly",
    Symbol(message): "{\\"level\\":\\"silly\\",\\"message\\":\\"call 6\\"}",
  },
]
`;
